### ISSUE TITLE
AFNetworking 3.0

### ISSUE THREAD

**Issue by JW97** [AFNetworking/AFOAuth2Manager#109]
Upon trying to update my project to use AFNetworking in CocoaPods I was greeted with:

'[!] Unable to satisfy the following requirements:

- `AFNetworking/NSURLConnection (~> 2.2)` required by `AFOAuth2Manager (2.1.0)`
- `AFNetworking (~> 3.0)` required by `Podfile`
- `AFNetworking (~> 2.2)` required by `AFOAuth2Manager (2.0.0)`

Specs satisfying the `AFNetworking/NSURLConnection (~> 2.2)` dependency were found, but they required a higher minimum deployment target.'

Any plans to update it?

**Comment by enwar3** [comment]
Having this same problem right now as well

**Comment by jameshays** [comment]
I see there is a branch for 3.0.0, so it appears support is coming.

**Comment by papley** [comment]
Any advice on whether the 3.0.0 branch is reasonably stable?

**Comment by skerkewitz** [comment]
Any news? It's been almost 2 month now...

**Comment by iulery** [comment]
:thumbsup:

**Comment by masuhara** [comment]
I have same issue... is this already resolved? or is there any way to avoid that with AFNetworking 3.0.x ?

**Comment by imadeit** [comment]
I have the same issue.
Any update plan towards AFNetworking 3.0.4?

**Comment by sushijani** [comment]
To give you an idea on stability of this branch with AFNetworking 3.0, a while back I modified the AFOAuth2Manager master branch code to work with AFNetworking 3.0. It was very straightforward and doable in matter of minutes. What I did:

- AFOAuth2Manager class inherits from AFHTTPSessionManager
- initWithBaseURL function: added 'sessionConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]' to the function call
- authenticateUsingOAuthWithURLString function: made the POST request compliant with AFHTTPSessionManager, as described in AFNetworking 3.0 migration guide. 
- authenticateUsingOAuthWithURLString function: changed the return types to NSURLSessionDataTask

That's all, the changes are akin to this 3.0.0 branch code. I can confirm it works rock solid with AFNetworking 3.0 with these changes. 

My point here is that if you are looking for compliance with the latest AFNetworking, the current AFOAuth2Manager master branch is very straightforward to modify to work with AFNetworking 3.0. I hope this is of help to some of you :)
